frontend/packages/client/src/components/ProposalsList/ProposalCardFooter.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useMemo, useState } from "react";
+import millify from "millify";
+import { Active, CheckCircle } from "../Svg";
+import { parseDateFromServer } from "utils";
+import StatusLabel from "../StatusLabel";
+import { useVotingResults } from "hooks";
+import { FilterValues } from "const";
+
+const ProposalCardFooter = ({ id, voted, endTime, computedStatus, isDesktopOnly }) => {
+  const [justMounted, setJustMounted] = useState(true);
+
+  const { getVotingResults, data: votingResults } = useVotingResults();
+
+  const { diffDuration } = parseDateFromServer(endTime);
+
+  const status = FilterValues[computedStatus] ?? FilterValues.closed;
+
+  useEffect(() => {
+    if (justMounted && (FilterValues[computedStatus] === FilterValues.closed)) {
+      async function _getVotingResults() {
+        return getVotingResults(id);
+      }
+      _getVotingResults();
+    }
+  }, [computedStatus, getVotingResults, id, justMounted]);
+
+  // prevents memory leak in above useEffect for getVotingResults
+  useEffect(() => () => { setJustMounted(false) }, []);
```

Comment:
Open to better approaches, but was unable to resolve the memory leak without this.

Reply:
maybe triggering only by checking votingResults has something?

Reply:
There is no need to wrap the custom hook for fetching voting results with useEffect. If you look at how `useVotingResults` is used in [ProposalInformation.js](https://github.com/DapperCollectives/CAST/blob/main/frontend/packages/client/src/components/ProposalInformation.js#L166-L167) you can just the following here to achieve the desired result:

```
const { data: votingResults } = useVotingResults(id);
```